viadot/sources/sharepoint.py

```diff
@@ -65,3 +69,29 @@ def download_file(
             url=download_from_path,
             filename=download_to_path,
         )
+
+    def download_file_from_url(self, rel_file_url, download_folder: str = "./") -> None:
```

Comment:
Delete the previous method `download_file`. Add an option to provide a file name not only download_folder.
You get the filename on line 89, but it should be possible to provide that name from flow/task. Check the old method (lines 55-56)